types/swagger-parser/index.d.ts

```diff
@@ -1,7 +1,50 @@
 // Type definitions for swagger-parser 4.x
 // Project: https://www.npmjs.com/package/swagger-parser
 // Definitions by: Tobias Wolff <https://github.com/Tobias4872/>
+//                 Eric Henry <https://github.com/EricHenry/>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-export function parse(api: any, options?: any): Promise<any>;
-export function parse(api: any, options: any, callback: (err: any, result?: any) => void): void;
+export namespace SwaggerParser {}
+
+export interface Options {
+    allow?: {
+        json?: boolean;
+        yaml?: boolean;
+        empty?: boolean;
+        unknown?: boolean;
+    };
+    $ref?: {
+        internal?: boolean;
+        external?: boolean;
+        circular?: boolean | 'ignore';
+    };
+    validate?: {
+        schema?: boolean;
+        spec?: boolean;
+    };
+    cache?: {
+        fs?: number;
+        http?: number;
+        https?: number;
+    };
+}
+
+export function parse(api: string | any, options?: Options): Promise<any>;
```

Comment:
```string | any``` is rather misleading, as it's the same exact type as ```any``` so it doesn't provide any stronger type checking.

Furthermore the docs state this should be a [SwaggerObject](https://github.com/BigstickCarpet/swagger-parser/blob/master/docs/swagger-parser.md#parseapi-options-callback) 

How about defining a ```Swagger``` interface and having the argument be ```string | Swagger``` ? 

If you'd rather not do that, then I'd say convert all the ```string | any```s to ```any```s.